django_auth_ldap/backend.py

```diff
@@ -833,6 +858,12 @@ def _bind_as(self, bind_dn, bind_password, sticky=False):
         the life of this object. If False, then the caller only wishes to test
         the credentials, after which the connection will be considered unbound.
         """
+
+        if self.settings.BIND_SASL_MECHANISM and self._connection:
+            # Since an existing connection is likely to have used sasl
+            # and thus be unusable for binds, reset it first
+            self._connection = None
```

Comment:
This worked in my testing, (and in our deployments) but perhaps there's a better way?

Reply:
Why does the connection need to be reset? Can’t another binding be used over the same connection?

Reply:
I'm not sure why, but it definitely doesn't work when using SASL/GSSAPI in our environment. It might be better to use a separate connection for the authentication checks, unless someone specifies otherwise. In that case, we could remove this (though we'd need to change something else)

Reply:
Resetting connection to None is fairly expensive, it disconnects from the LDAP, an requires to establish a new TLS connection.

The LDAP protocol allows rebinding over an open connection. This is used to check the username and password provided during authentication. I’m curious why the rebinding fails when SASL is used? What behavior are you seeing?

I read on the Internet that some LDAP servers drop the connection abruptly on rebinds. Is that what you are experiencing?